src/radical/pilot/utils/component.py

```diff
@@ -335,18 +336,36 @@ def _control_cb(self, topic, msg):
 
         # try to handle message as RPC message
         try:
-            self._handle_zmq_msg(msg)
-            # handled successfully
-            return
+            msg = ru.zmq.Message.deserialize(msg)
+
+            if isinstance(msg, RPCRequestMessage):
+
+                self._log.debug_4('handle rpc request %s', msg)
+                self._handle_rpc_msg(msg)
+                return
+
+            elif isinstance(msg, RPCResultMessage):
+
+                if msg.uid in self._rpc_reqs:
+                    self._log.debug_4('handle rpc result %s', msg)
+                    self._rpc_reqs[msg.uid]['res'] = msg
+                    self._rpc_reqs[msg.uid]['evt'].set()
+
+                return
+
+            else:
+                raise ValueError('message type not handled')
+
+        except Exception as e:
 
-        except:
             # could not be handled - fall through to legacy handlers
-            pass
+            self._log.debug('fall back to legacy msg handlers [%s]', repr(e))
```

Comment:
`debug_4`? (too many messages started to appear in the regular debug mode)

Reply:
ack